### PR TITLE
KAFKA-17548: Improve group listing integration test

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala [hunk 1]
```diff
@@ -2147,15 +2147,17 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
           TestUtils.waitUntilTrue(() => {
             val matching = client.listConsumerGroups.all.get.asScala.filter(group =>
               group.groupId == testGroupId &&
-                group.state.get == ConsumerGroupState.STABLE)
+                group.state.get == ConsumerGroupState.STABLE &&
+                group.groupState.get == GroupState.STABLE)
```
Comment (apoorvmittal10) [inline core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2151]: Just a query: Why do we need to check `group.state.get` && `group.groupState.get` but just not `group.groupState.get` as the other method is deprecated and uses groupState itself internally?

**Review by apoorvmittal10 — APPROVED** [review]
LGTM! A query to understand the reason to check the state with deprectaed method.

File: core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala (hunk 1 above)
Comment (AndrewJSchofield) [reply core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2151]: I want to ensure that they give the same result and continue to give the same result in the future. I know it should be obvious that they do, but a lot of testing is like that. In 5.0, we'll remove `ConsumerGroupState` entirely.

**Review by omkreddy — APPROVED** [review]
@AndrewJSchofield Thanks for the PR. LGTM

File: core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala (hunk 1 above)
Comment (apoorvmittal10) [reply core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2151]: Thanks for explaining.

**Pull request merged**